### PR TITLE
[WIP] enable step3p5 for bf16 and fp8

### PR DESCRIPTION
**Pull request by ranzhejiang**

### CODE REVIEW

File: vllm_gaudi/ops/hpu_fp8.py [hunk 1]
```diff
@@ -223,10 +223,27 @@ def apply_monolithic(
             topk_weights, topk_ids = layer.router.select_experts(hidden_states=x, router_logits=router_logits)
         else:
             import torch.nn.functional as F
-            topk_weights = F.softmax(router_logits, dim=1, dtype=torch.float32)
-            topk_weights, topk_ids = torch.topk(topk_weights, layer.top_k, dim=-1)
-            topk_weights /= topk_weights.sum(dim=-1, keepdim=True)
-            topk_weights = topk_weights.to(x.dtype)
+            scoring_func = getattr(layer, "scoring_func", "softmax")
+            if scoring_func == "sigmoid":
+                # print("ssssssssssss")
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/ops/hpu_fp8.py]: delete it

File: vllm_gaudi/extension/ops.py [hunk 2]
```diff
@@ -670,8 +682,120 @@ def forward(self, hidden_states, expert_routing_table, router_weights, permuted_
 
         w13_list = self._cached_w13_views
         w2_list = self._cached_w2_views
+        use_loop = False
+        if(activation != "silu" and use_loop):
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/extension/ops.py]: use real actn_func like activation == "SwigluStepAndMul", not != "silu"

File: vllm_gaudi/extension/ops.py [hunk 3]
```diff
@@ -670,8 +682,120 @@ def forward(self, hidden_states, expert_routing_table, router_weights, permuted_
...
+            limit = 7.0
+            device = hidden_states.device
+            dtype = hidden_states.dtype
+            T, H = hidden_states.shape
+
+            # 本地 experts 数
+            E_local = len(self.w13_list)
+            if(activation != "silu"):
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/extension/ops.py]: same here

File: vllm_gaudi/extension/ops.py [hunk 4]
```diff
@@ -670,8 +682,120 @@ def forward(self, hidden_states, expert_routing_table, router_weights, permuted_
...
+                for local_e in range(E_local):
+                    global_eid = self.experts_min + local_e
+
+                    W13 = w13_list[local_e]  # 期望 [2D, H] 或等价
+                    W2  = w2_list[local_e]   # 期望 [H, D] 或等价
+
+                    gate_up = hidden_states @ W13.t()  # [T, 2D]
+
+                    # if activation == "silu":
+                    #     # 标准 swiglu: silu(gate) * up
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/extension/ops.py]: clean all the Chinese comments

File: vllm_gaudi/extension/ops.py [hunk 5]
```diff
@@ -670,8 +682,120 @@ def forward(self, hidden_states, expert_routing_table, router_weights, permuted_
...
+                    #     ff = F.silu(gate) * up
+                    # else:
+                    #     # 自定义 clamp 版本
+                    ff = self.custom_gateup_activation(gate_up, limit=limit)
+
+                    y = ff @ W2.t()  # [T, H]
+
+                    # 乘该 expert 的权重并累加
+                    out = out + y * experts_mask[global_eid]  # [T,H] * [T,1]
+        elif(activation != "silu" and not use_loop):
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/extension/ops.py]: same here

File: vllm_gaudi/extension/ops.py [hunk 6]
```diff
@@ -670,8 +682,120 @@ def forward(self, hidden_states, expert_routing_table, router_weights, permuted_
...
+            device = hidden_states.device
+            dtype = hidden_states.dtype  # expected bf16
+
+            limit = 7.0
+
+            # ---- local experts
+            E_local = len(self.w13_list)
+
+            # ------------------------------------------------------------
+            # 1) 准备权重：从 list 取出并 stack 成 batch
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/extension/ops.py]: clean them

File: vllm_gaudi/extension/ops.py [hunk 7]
```diff
@@ -670,8 +682,120 @@ def forward(self, hidden_states, expert_routing_table, router_weights, permuted_
...
+            W2 = self.w2_raw
+
+            twoD = W13.size(1)
+            if twoD % 2 != 0:
+                raise RuntimeError(f"W13 second dim should be 2D and even, got {twoD}")
+            D = twoD // 2
+
+            # ------------------------------------------------------------
+            # 2) 构造 experts_mask_local: [E_local, T, 1]
+            #    这里完全沿用你原写法：先做 global mask，再按 global_eid 切本地段
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/extension/ops.py]: clean them

File: vllm_gaudi/extension/ops.py [hunk 8]
```diff
@@ -1275,8 +1458,212 @@ def forward(
...
+                B_scale_inv=W2s,
+                bias=None,
+                accumulate=False,
+            )  # expect [E,T,H]
+
+            # -------------------------
+            # route-weight + sum -> [T,H] (slice method, no index_select)
+            # -------------------------
+            out = (y * experts_mask_local).sum(dim=0)  # [T,H]
+        elif(activation != "silu" and use_loop):
```
Comment (Wei-Lin-Intel) [inline vllm_gaudi/extension/ops.py]: If bmm path does not have any accuracy issue, then we do not need use_loop path. Also change the condition.
